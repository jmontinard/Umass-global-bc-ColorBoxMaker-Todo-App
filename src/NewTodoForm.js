import React, { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';

function NewTodoForm({ addTodo }) {
  const [task, setTask] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo({ id: uuidv4(), task });
    setTask('');
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={task}
        onChange={(e) => setTask(e.target.value)}
        placeholder="Enter a new todo"
      />
      <button type="submit">Add Todo</button>
    </form>
  );
}

export default NewTodoForm;