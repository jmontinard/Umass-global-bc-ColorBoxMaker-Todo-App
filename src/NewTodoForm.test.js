import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

test('renders NewTodoForm without crashing', () => {
  const { getByPlaceholderText } = render(<NewTodoForm addTodo={() => {}} />);
  const inputElement = getByPlaceholderText('Enter a new todo');
  expect(inputElement).toBeInTheDocument();
});

test('matches snapshot for NewTodoForm', () => {
  const { asFragment } = render(<NewTodoForm addTodo={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});
