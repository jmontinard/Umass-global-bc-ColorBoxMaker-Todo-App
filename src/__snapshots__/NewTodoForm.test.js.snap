// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for NewTodoForm 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="Enter a new todo"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
